File: API_Contracts/Models/RequestModels/RequestDetailsModel.cs
using System;
using System.Collections.Generic;
using API_Contracts.Models.CommentModels;
using API_Contracts.Models.UserModels;

namespace API_Contracts.Models.RequestModels
{
    public class RequestDetailsModel : RequestModel
    {
        public string Id { get; set; }
        public DateTime CreationDate { get; set; }
        public UserInfoModel UserInfo { get; set; }
        public IEnumerable<CommentDashboardModel> Comments { get; set; }
    }
}

File: API/ClientApp/src/app/main/search/request-details/request-details.model.ts
import { CommentDashboardModel } from "./Comment-dashboard.model";
import { RequestModel } from "../../request-creator/request.model";
import { UserInfoModel } from "../../account/userInfo.model";

export class RequestDetailsModel extends RequestModel {
    id: string;
    creationDate: Date;
    userInfo: UserInfoModel;
    comments: CommentDashboardModel[];
}
